src/client/dfs/dfs.c

```diff
@@ -1075,12 +1075,13 @@ open_symlink(dfs_t *dfs, dfs_obj_t *parent, int flags, daos_oclass_id_t cid,
 		entry->value_len = value_len;
 		rc = insert_entry(parent->oh, DAOS_TX_NONE, sym->name, len,
 				  DAOS_COND_DKEY_INSERT, entry);
-		if (rc) {
+		if ( (rc == -DER_EXIST) || (rc==0) ) {
```

Comment:
(style)  spaces required around that '==' (ctx:VxV)
(style)  space prohibited after that open parenthesis '('
(style)  space prohibited before that close parenthesis ')'